lib/storage/src/content_manager/snapshots.rs

```diff
@@ -0,0 +1,85 @@
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+
+use collection::operations::snapshot_ops::{
+    get_snapshot_description, list_snapshots_in_directory, SnapshotDescription,
+};
+use tar::Builder as TarBuilder;
+use tokio::io::AsyncWriteExt;
+
+use crate::content_manager::toc::FULL_SNAPSHOT_FILE_NAME;
+use crate::{StorageError, TableOfContent};
+
+pub async fn get_full_snapshot_path(
+    toc: &TableOfContent,
+    snapshot_name: &str,
+) -> Result<PathBuf, StorageError> {
+    let snapshot_path = Path::new(toc.snapshots_path()).join(snapshot_name);
+    if !snapshot_path.exists() {
+        return Err(StorageError::NotFound {
+            description: format!("Snapshot {} not found", snapshot_name),
+        });
+    }
+    Ok(snapshot_path)
+}
+
+pub async fn do_list_full_snapshots(
+    toc: &TableOfContent,
+) -> Result<Vec<SnapshotDescription>, StorageError> {
+    let snapshots_path = Path::new(toc.snapshots_path());
+    Ok(list_snapshots_in_directory(snapshots_path).await?)
+}
+
+pub async fn do_create_full_snapshot(
```

Comment:
Should we capture the `aliases` as well in that case?

Reply:
hmm, yes. Will do

Reply:
Updated